File: programs/assignment1/ProgrammingAssignment1/src/programmingassignment1/Doctor.java
// COSC 1320 Summer 2015
// Name: Adrian
// Programming Assignment 1
// This is my own work; I will not post

package programmingassignment1;

/**
 * 
 * @author Adrian Davila
 */
public class Doctor extends Employee{
    String specialty;

    Doctor(){
        specialty = "";
    }
    Doctor(char newRole , String newName , int newID , String newSpec){
        super(newRole , newName , newID);
        specialty = newSpec;
        addEmployee((Employee)this);
    }
    // Mutators
    public void deleteEmployee(){
        deleteEmployee(getName() , getRole());
    }
    public void setSpec(String newSpec){
        specialty = newSpec;
    }
    // Accessors
    public String getSpec(){
        return specialty;
    }
    @Override
    public String toString(){
        return (super.toString() + " " + specialty);
    }
    @Override
    public String toString(boolean t){
        return (super.toString() + " Specialty: " + specialty);
    }
    @Override
    public boolean equals(Employee obj){
        if(obj == null){
            return false;
        }else if(getClass() != obj.getClass()){
            return false;
        }else{
            Doctor otherDoc = (Doctor)obj;
            return (getName().equals(otherDoc.getName()) && getRole() == otherDoc.getRole());
        }
    }
}

File: programs/assignment3/ProgrammingAssignment3/ProgrammingAssignment3/Doctor.cpp
// COSC 1320 Summer 2015
// Name: Adrian
// Programming Assignment 3
// This is my own work; I will not post

#include "Doctor.h"
#include <iostream>
using namespace std;


Doctor::Doctor(void): HospitalEmployee(){
	specialty = "none";
}

Doctor::Doctor(char newRole, string newName, int newID, string newSpec) : HospitalEmployee(newRole , newName , newID){
	specialty = newSpec;
	//cout << "doc hit: " << specialty << endl;
}

Doctor::~Doctor(void)
{
}
string Doctor::getSpecialty(){
	return specialty;
}
void Doctor::setSpecialty(string newSpec){
	specialty = newSpec;
}
string Doctor::toString(){
	string str = "";
	str += HospitalEmployee::toString();
	str += " " + specialty;
	//cout << "spec: " << str << endl;
	return (str);
}
bool Doctor::equals(HospitalEmployee obj){
	if(obj.getID() == getID() && obj.getName() == getName() && obj.getRole() == getRole()){
		return true;
	} else {
		return false;
	}
}